types/wordpress__block-editor/components/block-controls.d.ts

```diff
@@ -2,8 +2,16 @@ import { Slot, Toolbar } from "@wordpress/components";
 import { FC, JSX, ReactNode } from "react";
 
 declare namespace BlockControls {
+    type BlockControlGroup =
+        | "default"
+        | "block"
+        | "inline"
+        | "other"
+        | "parent";
+
     interface Props extends Pick<Toolbar.Props, "controls"> {
         children: ReactNode;
+        group?: BlockControlGroup;
```

Comment:
I wouldn't call this a requirement, but you can change your optionals to use `| undefined` for the `exactOptionalPropertyTypes` config option: <https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes>

```suggestion
        group?: BlockControlGroup | undefined;
```

(This applies to other optional properties, not just this one)

Reply:
Yeah happy to. Any reason why? I thought `?` added `undefined` to the set. Not pushing back on the req, just curious. Thanks!

Reply:
There are a few cases in JavaScript where there's actually a difference between explicitly passing `undefined`, and just not providing a value at all. You can take a look at the page I linked if you're interested in reading a bit more about it: <https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes>

Reply:
Appreciate it.